linker error in u-boot with gcc 10.

First off, thanks for creating this!

I was trying it out (going to take a stab at adding support for #1), and I got a build error in u-boot when compiling v0.1.1:

```
/usr/bin/ld: scripts/dtc/dtc-parser.tab.o:(.bss+0x10): multiple definition of `yylloc'; scripts/dtc/dtc-lexer.lex.o:(.bss+0x0): first defined here                                            
collect2: error: ld returned 1 exit status                                                                                                                                                    
make[4]: *** [scripts/Makefile.host:106: scripts/dtc/dtc] Error 1                                                                                                                             
make[3]: *** [scripts/Makefile.build:432: scripts/dtc] Error 2                                                                                                                                
make[2]: *** [Makefile:515: scripts] Error 2                                                                                                                                                  
make[1]: *** [package/pkg-generic.mk:250: /home/andrew/repos/nerves_system_npi_imx6ull/.nerves/artifacts/nerves_system_npi_imx6ull-portable-0.1.1/build/uboot-d447c4866200ae5fd069313d92e2308f
f01af021/.stamp_built] Error 2                                                                                                                                                                
make: *** [Makefile:23: _all] Error 2                                                                                                                                                         
```

After some googling, I found that gcc 10 defaults to -fno-common, which causes the above error during linking u-boot. It's been fixed in upstream u-boot, but not in the NPi i.MX6ULL branch. I've taken the patch from u-boot that fixes the issue: https://gitlab.denx.de/u-boot/u-boot/-/commit/018921ee79d3f30893614b3b2b63b588d8544f73 and added it to the uboot patches directory. If you'd rather handle it a different way, then let me know.

--- a/uboot/0001-Remove-redundant-YYLOC-global-declaration.patch
+++ b/uboot/0001-Remove-redundant-YYLOC-global-declaration.patch
@@ -1,0 +1,27 @@
+From 018921ee79d3f30893614b3b2b63b588d8544f73 Mon Sep 17 00:00:00 2001
+From: Peter Robinson <pbrobinson@gmail.com>
+Date: Thu, 30 Jan 2020 09:37:15 +0000
+Subject: [PATCH] Remove redundant YYLOC global declaration
+
+Same as the upstream fix for building dtc with gcc 10.
+
+Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
+---
+ scripts/dtc/dtc-lexer.l | 1 -
+ 1 file changed, 1 deletion(-)
+
+diff --git a/scripts/dtc/dtc-lexer.l b/scripts/dtc/dtc-lexer.l
+index fd825ebba6..24af549977 100644
+--- a/scripts/dtc/dtc-lexer.l
++++ b/scripts/dtc/dtc-lexer.l
+@@ -38,7 +38,6 @@ LINECOMMENT	"//".*\n
+ #include "srcpos.h"
+ #include "dtc-parser.tab.h"
+ 
+-YYLTYPE yylloc;
+ extern bool treesource_error;
+ 
+ /* CAUTION: this will stop working if we ever use yyless() or yyunput() */
+-- 
+2.27.0
+

